src/elements/Table.tsx

```diff
@@ -2,10 +2,34 @@
 import MD, { Component, Fragment, MarkdownChildren, MarkdownElement } from "..";
 import { intersperse } from "../util/intersperse";
 
+export enum TableAlignment {
+  LEFT = "left",
+  CENTER = "center",
+  RIGHT = "right",
+}
+
+/** @internal */
+interface TableColumn {
+  width: number;
+  alignment?: TableAlignment;
+}
+
+/** @internal */
+interface TableHeader {
+  title: MarkdownChildren;
+  alignment?: TableAlignment;
+}
+
 /** @internal */
 interface Props<Headers extends string> {
   body: Record<Headers, MarkdownChildren>[];
-  headers: Record<Headers, MarkdownChildren>;
+  headers: Record<Headers, MarkdownChildren | TableHeader>;
+}
+
+/** @internal */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function isTableHeader(value: any): value is TableHeader {
+  return !!value.title;
```

Comment:
This will fail if `value` is `null` or `undefined`, best to add a check for that as well.